zeebe/zb-db/src/main/java/io/camunda/zeebe/db/impl/rocksdb/SnapshotOnlyDb.java

```diff
@@ -103,4 +112,55 @@ private UnsupportedOperationException unsupported(final String operation) {
         "Failed to execute 'ZeebeDb#%s'; this operation is not supported on a snapshot-only DB"
             .formatted(operation));
   }
+
+  public void copySnapshot(
+      final ZeebeTransactionDb<ZbColumnFamilies> toDB, final Set<ColumnFamilyScope> scopes) {
+    try (final var readOptions = PrefixReadOptions.readOptions()) {
+      final var toCtx = toDB.createContext();
+      final var abort = new AtomicBoolean(false);
+      for (final var cf : ZbColumnFamilies.values()) {
+        if (abort.get()) {
+          break;
+        }
+        if (!scopes.contains(cf.partitionScope())) {
+          continue;
+        }
+        LOG.debug("Copying column family '{}'", cf);
+        final var toCf = new RawTransactionalColumnFamily(toDB, cf);
+        final var cfContext = new ColumnFamilyContext(cf.getValue());
+        toCtx.runInTransaction(
+            () -> {
+              final var toTx = (ZeebeTransaction) toCtx.getCurrentTransaction();
+              cfContext.withPrefixKey(
+                  new DbNullKey(),
+                  (prefixKey, prefixLength) -> {
+                    try (final RocksIterator iterator = db.newIterator(readOptions)) {
+                      RawTransactionalColumnFamily.forEachPreallocated(
+                          iterator,
+                          cf,
+                          prefixKey,
+                          0,
+                          prefixLength,
+                          (key, keyOffset, keyLen, value, valueOffset, valueLen) -> {
+                            try {
+                              toCf.rawPut(
+                                  toTx, key, keyOffset, keyLen, value, valueOffset, valueLen);
+                            } catch (final Exception e) {
+                              LOG.error(
+                                  "Failed to copy column family '{}' on key {} and value with length {} terminating.",
+                                  cf,
+                                  new String(key),
+                                  value.length);
+                              LOG.error("Exception", e);
```

Comment:
```suggestion
                              LOG.error(
                                  "Failed to copy column family '{}' on key {} and value with length {} terminating.",
                                  cf,
                                  new String(key),
                                  value.length, e);
```

Reply:
just realized I also needed to add `new String(key, 0, keyLen)`